lucene/core/src/java/org/apache/lucene/store/BufferedIndexInput.java

```diff
@@ -154,65 +144,60 @@ public final void readBytes(byte[] b, int offset, int len, boolean useBuffer) th
         // this function, there is no need to do a seek
         // here, because there's no need to reread what we
         // had in the buffer.
-        long after = bufferStart+bufferPosition+len;
+        long after = bufferStart+buffer.position()+len;
         if(after > length())
           throw new EOFException("read past EOF: " + this);
-        readInternal(b, offset, len);
+        readInternal(ByteBuffer.wrap(b, offset, len));
         bufferStart = after;
-        bufferPosition = 0;
-        bufferLength = 0;                    // trigger refill() on read
+        buffer.position(0);
+        buffer.limit(0);  // trigger refill() on read
       }
     }
   }
 
   @Override
   public final short readShort() throws IOException {
-    if (2 <= (bufferLength-bufferPosition)) {
-      return (short) (((buffer[bufferPosition++] & 0xFF) <<  8) |  (buffer[bufferPosition++] & 0xFF));
+    if (Short.BYTES <= buffer.remaining()) {
+      return buffer.getShort();
```

Comment:
Can't remember off the top of my head - the byte order is always guaranteed here, right? Or should we assert the buffer has the right one?

Reply:
Yes: "Specific byte orders are represented by instances of the ByteOrder class. The initial order of a byte buffer is always BIG_ENDIAN." (https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/ByteBuffer.html)

Reply:
I think it's still a good call to add assertions to make sure that sub classes don't mess up with the ByteBuffer. I'll add some.

Reply:
+1. I checked the limit/remaining and other calls but the correctness there seems to be better verified by tests than by hand. :) Overall, like I said - looks like a nice simplification to me.